pkg/config/log.go

```diff
@@ -339,8 +346,82 @@ func extractShortPathFromFullPath(fullPath string) string {
 	return slices[len(slices)-1]
 }
 
+func createExtraJSONContext(params string) seelog.FormatterFunc {
+	return func(message string, level seelog.LogLevel, context seelog.LogContextInterface) interface{} {
+		contextList, ok := context.CustomContext().([]interface{})
+		if len(contextList) == 0 || !ok {
+			return ""
+		}
+		return extractContextString(jsonFormat, contextList)
+	}
+}
+
+func createExtraTextContext(params string) seelog.FormatterFunc {
+	return func(message string, level seelog.LogLevel, context seelog.LogContextInterface) interface{} {
+		contextList, ok := context.CustomContext().([]interface{})
+		if len(contextList) == 0 || !ok {
+			return ""
+		}
+		return extractContextString(textFormat, contextList)
+	}
+}
+
+func extractContextString(format contextFormat, contextList []interface{}) string {
+	builder := strings.Builder{}
+	if len(contextList) > 0 && format == jsonFormat {
+		builder.WriteString(",")
+	}
+
+	for i := 0; i < len(contextList); {
+		if i == len(contextList)-1 {
+			// key without value
+			return ""
+		}
+		key, val := contextList[i], contextList[i+1]
+		keyStr, ok := key.(string)
+		if !ok {
+			// skip non string keys
+			i += 2
+			continue
+		}
+		addToBuilder(&builder, keyStr, val, format, i == len(contextList)-2)
+		i += 2
+	}
+	return builder.String()
+}
+
+func addToBuilder(builder *strings.Builder, key string, value interface{}, format contextFormat, isLast bool) {
+	builder.WriteString(quoteJSON(format, key) + ":")
+	switch val := value.(type) {
+	case string:
+		builder.WriteString(quoteJSON(format, val))
+	case int:
+		builder.WriteString(strconv.Itoa(val))
+	case float64:
+		builder.WriteString(strconv.FormatFloat(val, 'f', 4, 32))
+	default:
+		builder.WriteString(quoteJSON(format, "non supported field"))
+	}
+	if !isLast {
+		if format == jsonFormat {
+			builder.WriteString(",")
+		} else {
+			builder.WriteString(" ")
+		}
+	}
+}
+
+func quoteJSON(format contextFormat, s string) string {
+	if format == jsonFormat {
```

Comment:
you could cut down on an allocation-per-field by using a shared buffer for creating the quoted string:


```
// in add-to-builder
var buf []byte


func appendFmt(builder StringBuilder, format contextFormat, s string, buf []byte)  []byte {
   if format == jsonFormat {
        buf = buf[:]
        strconv.AppendQuote(buf, s)
       builder.WriteBytes(buf)
   } else {
       builder.WriteString(s)
   }
} 
```

Reply:
My 2 cents are that we shouldn't make hypothetical performance improvements without seeing actual numbers from a benchmark, regardless of this has been proven from previous experience.

@leeavital also, in your example, it's ambiguous what the `buf` variable represents, since you are shadowing a global variable in your function code.

Reply:
@gbbr I definitely agree,  this would only be worth doing if it showed an improvement in the benchmarks that are referenced in the commit description.

They are shadowed intentionally, `buf` would be declared in addToBuilder and addToBuilder would call `appendFmt` once per arg using the same `buf`

Reply:
Thanks @leeavital I didn't know `strconv.AppendQuote` but this allowed to gain a few allocation and looks like it's a bit faster as well:

Base:
```
BenchmarkLogFormatWithContextFormatting-4      	  500000	      3718 ns/op	    1388 B/op	      34 allocs/op
```

With Change:
```
BenchmarkLogFormatWithContextFormatting-4      	  500000	      3376 ns/op	    1348 B/op	      29 allocs/op
```